server/src/test/java/org/opensearch/indices/IndicesRequestCacheTests.java

```diff
@@ -598,9 +551,8 @@ public void testStaleCount_OnRemovalNotificationOfStaleKey_DoesNotDecrementsStal
         // cache count should not be affected
         assertEquals(2, cache.count());
 
-        OpenSearchDirectoryReader.DelegatingCacheHelper delegatingCacheHelper = (OpenSearchDirectoryReader.DelegatingCacheHelper) reader
-            .getReaderCacheHelper();
-        String readerCacheKeyId = delegatingCacheHelper.getDelegatingCacheKey().getId();
+        // evict entry from second reader (this reader is not closed)
+        String readerCacheKeyId = getReaderCacheKeyId(secondReader);
```

Comment:
We should verify the state of `cleanupKeyToCountMap` as well in all the tests.

Reply:
I have a seperate test suite for this `testCleanupKeyToCountMapAreSetAppropriately`